sdks/go/pkg/beam/core/runtime/xlangx/expansionx/process_test.go

```diff
@@ -0,0 +1,73 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expansionx
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+func TestNewExpansionServiceRunner(t *testing.T) {
+	testPath := "path/to/jar"
+	serviceRunner := NewExpansionServiceRunner(testPath)
+	if serviceRunner.jarPath != testPath {
+		t.Errorf("JAR path mismatch: wanted %v, got %v", testPath, serviceRunner.jarPath)
+	}
+	commandString := strings.Join(serviceRunner.serviceCommand.Args, " ")
+	expCommandString := "java -jar " + testPath + " 8097"
+	if commandString != expCommandString {
+		t.Errorf("command mismatch: wanted %v, got %v", expCommandString, commandString)
+	}
+}
+
+func TestStartService_badCommand(t *testing.T) {
+	serviceRunner := NewExpansionServiceRunner("")
+	serviceRunner.serviceCommand = exec.Command("jahva", "-jar")
+	err := serviceRunner.StartService()
+	if err == nil {
+		t.Error("StartService succeeded when it should have failed")
+	}
+}
+
+func TestStartService_good(t *testing.T) {
+	testPath := "path/to/jar"
+	serviceRunner := NewExpansionServiceRunner(testPath)
+	serviceRunner.serviceCommand = exec.Command("which", "go")
```

Comment:
A suggestion in case you want to test these without using actual commands: Create an interface that's fulfilled by exec.Cmd containing the methods you need to use. That way you can create a mock version of that interface for testing purposes, just to make sure it gets properly started and stopped.

That seems like a bit too much work for this relatively bare-bones class at the moment. But if your code starts to get more complex and you need a way to test it, that's what I would recommend.

Reply:
I'm still getting used to considering interfaces as a way to enable mocking/dependency injection for testing. I agree that it seems like overkill for this class, but definitely a tool to keep in mind moving forward